test/box-luatest/interval_test.lua

```diff
@@ -0,0 +1,137 @@
+local server = require('test.luatest_helpers.server')
+local t = require('luatest')
+local g = t.group()
+
+g.before_all = function()
+    g.server = server:new({alias = 'master'})
+    g.server:start()
+end
+
+g.after_all = function()
+    g.server:stop()
+end
+
+g.test_interval_create_space = function()
+    g.server:exec(function()
+        local t = require('luatest')
+        local format = {{name = "i", type = "integer"},
+                        {name = "a", type = "interval"}}
+        local s = box.schema.space.create('S1', {format = format})
+        t.assert_equals(s ~= nil, true)
+        t.assert_equals(s:format(), format)
+        s:drop()
+    end)
+end
+
+g.test_interval_create_index = function()
+    g.server:exec(function()
+        local t = require('luatest')
+        local format = {{'i', 'integer'}, {'a', 'interval'}}
+        local parts = {{'a', 'interval'}}
+        local s = box.schema.space.create('S2', {format = format})
+        local _, err = pcall(s.create_index, s, 'a', {parts = parts})
+        local res = "Can't create or modify index 'a' in space 'S2': "..
+                    "field type 'interval' is not supported"
+        t.assert_equals(err.message, res)
+        s:drop()
+    end)
+end
+
+g.test_interval_insert = function()
+    g.server:exec(function()
+        local t = require('luatest')
+        local msgpack = require('msgpack')
+        local valbin = '\xc7\x07\x06\x03\x00\x01\x03\x7f\x05\x0f'
```

Comment:
Can't you use `interval.new()`? Same in the other places below.

Reply:
Fixed. However, we can't compare two intervals directly, so I still use `msgpack.encode()` to compare them.

Reply:
Now we can use INTERVAL values in `assert_equals()`, so I dropped unnecessary code.